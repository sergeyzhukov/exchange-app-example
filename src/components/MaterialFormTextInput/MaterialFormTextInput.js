import React from 'react'
import { View, StyleSheet, Text, TextInput } from 'react-native'
import Colors from '../../utils/colors'

export default ({ input, title, meta, ...inputProps }) => {
  const { touched, error } = meta

  return (
    <View>
      {input.value.length > 0 && <Text style={styles.placeholder}>{title}</Text>}
      <TextInput
        {...inputProps}
        style={styles.input}
        onChangeText={input.onChange}
        onBlur={input.onBlur}
        onFocus={input.onFocus}
        value={input.value}
      />
      {touched && error && <Text style={styles.error}>{error}</Text>}
    </View>
  )
}

const styles = StyleSheet.create({
  placeholder: {
    color: Colors.BRAND,
    fontWeight: '500',
    fontSize: 13,
  },
  error: {
    fontSize: 13,
    color: 'red',
    marginTop: 4,
  },
  input: {
    paddingBottom: 6,
    paddingTop: 4,
    fontSize: 17,
    borderBottomColor: '#cccccc',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
})
