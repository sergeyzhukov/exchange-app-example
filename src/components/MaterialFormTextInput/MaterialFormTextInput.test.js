import React from 'react'
import renderer from 'react-test-renderer'
import MaterialFormTextInput from './MaterialFormTextInput'

test('render <MaterialFormTextInput />', () => {
  const input = { value: '' }
  const meta = { touched: false }

  const component = <MaterialFormTextInput input={input} meta={meta} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render <MaterialFormTextInput /> with some input ', () => {
  const input = { value: 'test' }
  const meta = { touched: true }

  const component = <MaterialFormTextInput input={input} meta={meta} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render <MaterialFormTextInput /> with error', () => {
  const input = { value: 'test' }
  const meta = { touched: true, error: 'Error' }

  const component = <MaterialFormTextInput input={input} meta={meta} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
