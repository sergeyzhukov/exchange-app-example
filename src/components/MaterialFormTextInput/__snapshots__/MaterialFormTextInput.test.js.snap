// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <MaterialFormTextInput /> 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    style={
      Object {
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 0.5,
        "fontSize": 17,
        "paddingBottom": 6,
        "paddingTop": 4,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`render <MaterialFormTextInput /> with error 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#4072B8",
        "fontSize": 13,
        "fontWeight": "500",
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    style={
      Object {
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 0.5,
        "fontSize": 17,
        "paddingBottom": 6,
        "paddingTop": 4,
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 13,
        "marginTop": 4,
      }
    }
  >
    Error
  </Text>
</View>
`;

exports[`render <MaterialFormTextInput /> with some input  1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#4072B8",
        "fontSize": 13,
        "fontWeight": "500",
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    style={
      Object {
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 0.5,
        "fontSize": 17,
        "paddingBottom": 6,
        "paddingTop": 4,
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
</View>
`;
